### DISCUSSION TITLE
Switch tile to Master tile

### DISCUSSION THREAD

**Discussion by h8uthemost** [i3/i3#4692] [Q&A]
I'm looking through the config file and I can't find how to switch a focused to the left master window. Or top master window. Wherever I put the master. Kind of like how Spectrwm and DWM has. Does this exist in the manual tiling? Because sometimes I get lost when trying to manually put a tile into the master using Mod+ Direction keys.

Thank you for any and all help.

**Comment by Airblader** [comment]
You can use the `swap` command for this, possibly in combination with window marks. See user guide for both of these features. Likely this requires some scripting to make it comfortably work how you want. 

The concept of a layout with a "master" doesn't exist in i3.

**Comment by vincentbernat** [comment]
You can check this [blog post](https://aduros.com/blog/hacking-i3-window-promoting/) for a solution.

**Comment by h8uthemost** [comment]
That would be perfect, vincent! Except it doesn't work for me. :( It's either the lack of my repos not having i3ipc or the fact it doesn't mention a keybind. He mentions xmonad -> Master is Alt+Return, but that combo isn't working.

Damn so close. Thanks for the help though.

**Reply by vincentbernat** [reply]
Try running the script from a terminal first. The terminal should be promoted. Then, at the bottom of the post, it shows how to integrate into your i3 configuration and it's bound to mod+p.

**Reply by orestisfl** [reply]
Additionally, you can install [i3ipc](https://github.com/altdesktop/i3ipc-python) with pip

**Comment by h8uthemost** [comment]
I can't believe I didn't see the Mod+p. Thanks for that. And i3ipc-python is in the repos, I just spelt it wrong I guess.

Anyways it's still not working for me, here's the error terminal gives me when running the script through it:

`Traceback (most recent call last):
  File "/home/.scripts/i3master.sh", line 22, in <module>
    workspace = i3.get_tree().find_by_id(reply.ipc_data["id"])
AttributeError: 'WorkspaceReply' object has no attribute 'ipc_data'`

Not sure what I'm missing still. Not a huge deal as I'm getting better at "pulling" whatever window I want into the master stack. I just have to go around a couple times to get it there if that makes sense.

**Reply by vincentbernat** [reply]
What's your version of i3ipc Python? `WorkspaceReply` should have an `ipc_data` attribute since 2.2.1.

**Comment by h8uthemost** [comment]
2.2.1. That's what I got from pip
